include/triton/Dialect/Triton/IR/TritonOps.td

```diff
@@ -351,6 +351,9 @@ def TT_ReduceOp : TT_Op<"reduce", [NoSideEffect,
 
     let assemblyFormat = "$operand attr-dict `:` type($operand) `->` type($result)";
 
+    let extraClassDeclaration = [{
+        static bool withIndex(mlir::triton::RedOp redOp);
```

Comment:
Does it have to be static?

Reply:
This function is also used to determine the output element type **before** ReduceOp is constructed.

https://github.com/openai/triton/blob/qingyi/argmin-argmax/python/src/triton.cc#L1198

Reply:
This function is also used to determine the output element type **before** ReduceOp is constructed.

https://github.com/openai/triton/blob/qingyi/argmin-argmax/python/src/triton.cc#L1198